### PR TITLE
chore: Add release workflow

### PR DESCRIPTION
**Pull request by timfish**
Ref #122

`release-please.yml` is copied straight from here:
https://github.com/nodejs/node-addon-api/blob/main/.github/workflows/release-please.yml

For this to actually work it needs Github credentials that can create PRs that trigger other workflows. The default GITHUB_TOKEN cannot do this.
https://github.com/googleapis/release-please-action?tab=readme-ov-file#github-credentials

We also need an NPM token added to the secrets so it can publish.

### CODE REVIEW

**Comment by jsumners-nr** [comment]
The tool supports changing the log type to use the GitHub API - https://github.com/googleapis/release-please/blob/main/docs/customizing.md#changelog-types

**Comment by timfish** [comment]
> The tool supports changing the log type to use the GitHub API

Ah nice. Do you know where this setting goes? It's not immediately obvious!

**Review by jsumners-nr — APPROVED** [review]
Looks good to me.

**Comment by timfish** [comment]
Who do we need to ping to get the correct tokens added to the repo secrets? If we do that before this is merged it should create a release PR!

File: .github/workflows/release-please.yml
Comment (anonrig) [inline .github/workflows/release-please.yml:6]: This will release a new version on every push to main. Do we really want that, or should we go for a different approach?

**Comment by anonrig** [comment]
> Who do we need to ping to get the correct tokens added to the repo secrets? If we do that before this is merged it should create a release PR!

@bengl @mhdawson can you add the repo secret to land this?

File: .github/workflows/release-please.yml
Comment (timfish) [reply .github/workflows/release-please.yml:6]: This runs the action on every push to main but I don't think it will release every time. On every push to main it creates/updates a PR for the next release (like [this one](https://github.com/nodejs/node-addon-api/pull/1526)) and the release is only published once this PR is merged.

**Review by anonrig — APPROVED** [review]

**Comment by mhdawson** [comment]
@legendecas any chance you could help with adding the token since you did it recently for the node-addon-api repo and can probably do it faster than me?

**Comment by legendecas** [comment]
> any chance you could help with adding the token since you did it recently for the node-addon-api repo and can probably do it faster than me?

done. Added `NPM_TOKEN` repository secret granular token, expiring on July 31, 2025.

**Review by legendecas — APPROVED** [review]

**Comment by timfish** [comment]
Perfect, I think we just need a Github token now unless someone has done that?

[The docs](https://github.com/googleapis/release-please-action?tab=readme-ov-file#github-credentials) suggest the default included in GHA does not have the permissions to create PRs.

**Comment by legendecas** [comment]
`GITHUB_TOKEN` is automatically generated: https://docs.github.com/en/actions/security-guides/automatic-token-authentication.

I think this should be ready.

**Review by Qard — APPROVED** [review]

**Comment by timfish** [comment]
> GITHUB_TOKEN is automatically generated

It is automatically generated, but if PRs are created with that token, workflows don't run on them.

The [`release-please` docs](https://github.com/googleapis/release-please-action?tab=readme-ov-file#github-credentials) cover this and I've seen this issue on other repositories.

> By default, Release Please uses the built-in GITHUB_TOKEN secret. However, all resources created by release-please (release tag or release pull request) will not trigger future GitHub actions workflows, and workflows normally triggered by release.created events will also not run. 
>
> You will want to configure a GitHub Actions secret with a [Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) if you want GitHub Actions CI checks to run on Release Please PRs.

Githubs [own docs](https://docs.github.com/en/actions/using-workflows/triggering-a-workflow#triggering-a-workflow-from-a-workflow) also mention this:

> When you use the repository's `GITHUB_TOKEN` to perform tasks, events triggered by the `GITHUB_TOKEN` will not create a new workflow run. This prevents you from accidentally creating recursive workflow runs.

Although now I'm looking at [other](https://github.com/nodejs/node-addon-api/pull/1526) PRs [created](https://github.com/nodejs/node-gyp/pull/3006) by `release-please` in other Node org repos, workflows aren't running in those so maybe there's no need for workflows to run in the release PRs?

I'll merge this and it can be added later if it's a problem!

**Comment by timfish** [comment]
So the [PR has been created](https://github.com/nodejs/import-in-the-middle/pull/125) but the workflow hasn't run so there's no way to merge it!

**Pull request merged**
